### PR TITLE
Added functionality to make sure only e-mails from inbox are shown

### PR DESCRIPTION
**Pull request by Laskewitz**
#### Category
- [ ] Bug Fix
- [x] New Feature

#### What's in this Pull Request?

I added functionality to make sure only e-mails from the inbox are shown. It's a web part setting, so if you disable this in the edit web part view, you will see all messages (including sent ones).

### CODE REVIEW

File: solution/src/webparts/personalEmail/components/PersonalEmail.tsx [hunk 1]
```diff
@@ -34,8 +34,14 @@ export class PersonalEmail extends React.Component<IPersonalEmailProps, IPersona
       messages: []
     });
 
+    let graphURI: string = "me/messages";
+
+    if(this.props.showInboxOnly) {
```
Comment (waldekmastykarz) [inline solution/src/webparts/personalEmail/components/PersonalEmail.tsx]: Due to the comparison in line 95: (`prevProps.nrOfMessages !== this.props.nrOfMessages`), the data in the web part is not reloaded after changing the Inbox-only setting. We need to extend the check on line 95 to include checking if the value of the Inbox-only setting has changed and reload data if this is indeed the case.

**Comment by VesaJuvonen** [comment]
@Laskewitz - any change we could get the small adjustments on this as we would love to get your improvements on the actual solution?

**Comment by VesaJuvonen** [comment]
@Laskewitz - any change we could get the small adjustments on this as we would love to get your improvements on the actual solution? Thx.

**Comment by Laskewitz** [comment]
Fixed it.

**Comment by waldekmastykarz** [comment]
Thanks! I'll have a look shortly

**Comment by waldekmastykarz** [comment]
Sorry for the delay. Merged manually to resolve merge conflicts. Thanks @Laskewitz! 👏

**Pull request closed without merging**